### PR TITLE
Sub-daily Timeline

### LINKED ISSUE #23: Update react to the latest version
Looks like we are using 15.3.1 right now. Let's try updating to react 16.

### PR DESCRIPTION
**Pull request by ZachTRice**
See https://github.com/nasa-gibs/worldview/pull/778
Fixes #23

### CODE REVIEW

File: src/scripts/components/dateselector/wv.dateselector.js [hunk 1]
```diff
@@ -99,6 +107,46 @@ class dateSelector extends React.Component {
...
+            value={util.pad(this.state.date.getUTCHours(), 2, '0')}
+            tabIndex={4}
+            focused={(this.state.tab === 4)}
+            nextTab={this.nextTab.bind(this)}
+            maxDate={this.props.maxDate}
+            minDate={this.props.minDate}
+            blur={this.blur.bind(this)}
+          />
+        }
+        {this.state.maxZoom >= 4 && <div className="input-time-divider">:</div> }
```
Comment (Benjaki2) [inline src/scripts/components/dateselector/wv.dateselector.js]: In this case you could also pull this out into it's own function so it only has to run the logic `this.state.maxZoom >= 4` once. 

That would look something like:
```
  renderSubdaily() {
    if(this.state.maxZoom >= 4) {
      return ( 
        <div>
          <div className="input-time-divider">:</div>
          <DateInputColumn
            step={10}
            startDate={new Date(2000)}
            today={new Date()}
            date={this.state.date}
            type="minute"
            height={this.props.height}
            width={this.props.width}
            updateDate={this.updateDate.bind(this)}
            value={util.pad(this.state.date.getUTCMinutes(), 2, '0')}
            tabIndex={5}
            focused={(this.state.tab === 5)}
            nextTab={this.nextTab.bind(this)}
            maxDate={this.props.maxDate}
            minDate={this.props.minDate}
            blur={this.blur.bind(this)}
          />
          <div className="input-time-zmark">Z</div>
        </div>
      );
    }
  }
```

Both implementations work. Feel free to keep what you have or implement this. Either way.

File: src/scripts/components/dateselector/wv.dateselector.js [hunk 2]
```diff
@@ -99,6 +107,46 @@ class dateSelector extends React.Component {
           minDate={this.props.minDate}
           blur={this.blur.bind(this)}
         />
+        {this.state.maxZoom >= 4 &&
+          <DateInputColumn
```
Comment (Benjaki2) [inline src/scripts/components/dateselector/wv.dateselector.js]: Hooray for reusable components! `DateInputColumn`

File: src/scripts/components/dateselector/wv.dateselector.js [hunk 3]
```diff
@@ -22,22 +22,24 @@ class dateSelector extends React.Component {
...
+      maxZoom: props.maxZoom
     });
   }
   blur() {
     this.setState({tab: null});
   }
   nextTab(index) {
     var nextTab;
-    if (index < 3) {
+    if (index < 5) {
```
Comment (Benjaki2) [inline src/scripts/components/dateselector/wv.dateselector.js]: I think this `5` will need to be dynamic, right @ZachTRice ? could be passed as a tabLength prop based on maxZoom.

**Review by Benjaki2 — CHANGES_REQUESTED** [review]
Looks really good @ZachTRice ! 

I found one thing that might be a bug... And left a couple other comments

File: src/scripts/components/dateselector/wv.dateselector.js (hunk 1 above)
Comment (ZachTRice) [reply src/scripts/components/dateselector/wv.dateselector.js]: The initial reason I didn't do this is that react's wrapping div was affecting the tabbing. I have updated to react 16.2.0 which introduces fragments. So, I will give this a go using fragments.

File: src/scripts/components/dateselector/wv.dateselector.js (hunk 3 above)
Comment (ZachTRice) [reply src/scripts/components/dateselector/wv.dateselector.js]: 👍

Comment (ZachTRice) [reply src/scripts/components/dateselector/wv.dateselector.js]: I added a `maxTab` variable and set it based on the `maxZoom` state

File: src/scripts/components/dateselector/wv.dateselector.js (hunk 1 above)
Comment (ZachTRice) [reply src/scripts/components/dateselector/wv.dateselector.js]: I have added a `renderSubdaily()` method using `React.Fragment`. Unfortunately `eslint` has no update to handle the short `<>`, `</>` syntax so I had to write the long-hand `<React.Fragment>`, `</React.Fragment>`

**Comment by ZachTRice** [comment]
@Benjaki2 please review again.

**Review by Benjaki2 — APPROVED** [review]
Awesome work @ZachTRice !

I'm Fired up about the react migration 🎉

**Pull request merged**